module/VuFind/tests/unit-tests/src/VuFindTest/Record/LoaderTest.php

```diff
@@ -216,12 +303,35 @@ public function testBatchLoadWithFallback()
         $solrParams = new ParamBag();
         $solrParams->set('fq', 'id:test1');
 
-        $service = $this->createMock(\VuFindSearch\Service::class);
-        $service->expects($this->exactly(2))->method('retrieveBatch')
+        $commandObj = $this->getMockBuilder(\VuFindSearch\Command\AbstractBase::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $commandObj->expects($this->exactly(2))->method('getResult')
+            ->willReturnOnConsecutiveCalls($collection1, $collection2);
+
+        $service = $this->getMockBuilder(\VuFindSearch\Service::class)
+            ->disableOriginalConstructor()->getMock();
+
+        $checkCommand1  = function ($command) {
+            return get_class($command) === \VuFindSearch\Command\RetrieveBatchCommand::class
+                && ($command->getTargetIdentifier() === "Solr")
+                && $command->getArguments()[0] === ["test1", "test2"]
+                && ($command->getArguments()[1]->getArrayCopy() === ['fq'=> ['id:test1']]);
```

Comment:
Same comment here: might be worth checking all instances of getArrayCopy in this test and seeing if some/all can be eliminated.